src/Storages/ObjectStorage/S3/Configuration.cpp

```diff
@@ -732,7 +699,10 @@ void StorageS3Configuration::addStructureAndFormatToArgsIfNeeded(
             count -= key_value_asts.size();
         }
 
-        if (count == 0 || count > getMaxNumberOfArguments())
+        if (!count)
+            return;
```

Comment:
To be honest I'd still do `path` required, disks are not easy to add, so having a disk per table is not very realistic, but if `path` is not specified, then we cannot create more than one table on this disk and user will know know it until he does that and sees some vague exceptions or even incorrect result.

I just discussed this with Alexander Sapin, he said that his idea was that if `path` is empty, then we must use `root/<table_uuid>`, but never just `root`.